src/qibo/tensorflow/custom_operators/cc/kernels/measurements.cc

```diff
@@ -0,0 +1,120 @@
+#if GOOGLE_CUDA
+#define EIGEN_USE_GPU
+#endif  // GOOGLE_CUDA
+
+#include "measurements.h"
+#include "tensorflow/core/framework/op_kernel.h"
+
+namespace tensorflow {
+
+typedef Eigen::ThreadPoolDevice CPUDevice;
+typedef Eigen::GpuDevice GPUDevice;
+
+namespace functor {
+
+// CPU specialization
+template <typename Tint, typename Tfloat>
+struct MeasureFrequenciesFunctor<CPUDevice, Tint, Tfloat> {
+  void operator()(const CPUDevice &d, Tint* frequencies, const Tfloat* probs,
+                  Tint nshots, int nqubits, int user_seed = 1234)
+  {
+    int64 nstates = 1 << nqubits;
+    srand(user_seed);
+    unsigned thread_seed[omp_get_max_threads()];
```

Comment:
I would avoid using C99 specs here, I believe these `omp_*` functions return `int` not `const int`, so we may prefer to allocate dynamically or use a vector.